x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/rule_types/esql/utils/merge_esql_result_in_source.test.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { mergeEsqlResultInSource } from './merge_esql_result_in_source';
+
+describe('mergeEsqlResultInSource', () => {
```

Comment:
I don't see any examples of "object vs. primitive" conflicts, e.g.:

```ts
    const source = { 'agent': { name: 'hello' } };
    const esqlResult = { 'agent': 'primitive' };
```

Nor any tests demonstrating the preservation (or dropping) of keys, e.g.:

```ts
    const source = { 'log.syslog': { hostname: 'host-1', other: 'other' } };
    const esqlResult = { 'log.syslog.hostname': 'esql host' };
```

But please let me know if those are unexpected/impossible situations. I think the former _would_ represent a mapping conflict, but it also seems reasonable that you could e.g. `EVAL` a field whose value conflicts with the actual mapping? Food for thought.

Reply:
> I don't see any examples of "object vs. primitive" conflicts, e.g.:

    const source = { 'agent': { name: 'hello' } };
    const esqlResult = { 'agent': 'primitive' };

It's kind of possible, but as you suggested it's a mapping conflict. Resulted string field will be removed during fields and source merging https://github.com/elastic/kibana/tree/82c3b5b859e4f5d37bc74c23fe0ea022531d7a9d/x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/rule_types/utils/source_fields_merging to prevent failure during alert indexing

Reply:
I added a second test case https://github.com/elastic/kibana/pull/208894/commits/f2a96eb7485a78f56168c3e6256540dca730fee3